packages/date-time-utilities/src/dateGrid/DateGrid.ts

```diff
@@ -0,0 +1,184 @@
+import { IDay, IDayGridOptions, IRestrictedDatesOptions, IAvailableDateOptions, IGridStrings } from './DateGrid.types';
+import { addDays, getDateRangeArray, compareDates, isInDateRangeArray, compareDatePart } from '../dateMath/DateMath';
+import { find } from '@uifabric/utilities';
+import { DAYS_IN_WEEK, DateRangeType, DayOfWeek } from '../dateValues/DateValues';
+
+export const getDayGrid = (options: IDayGridOptions): IDay[][] => {
```

Comment:
One thing we need to resolve at some point is whether to use `const myFunc = () => { ... }` or `function myFunc() { ... }`. I very strongly prefer `function` (better for debugging, seems more semantically correct) and this has mostly been the convention in v7 aside from when creating React function components, but the convention in v0 appears to be using `const` for everything (I've tried to ask about this but nobody has ever explained why).